proto/pulumi/provider.proto

```diff
@@ -1012,39 +1036,40 @@ message ErrorResourceInitFailed {
 
 
 
-// GetMappingRequest allows providers to return ecosystem specific information to allow the provider to be
-// converted from a source markup to Pulumi. It's expected that provider bridges that target a given ecosystem
-// (e.g. Terraform, Kubernetes) would also publish a conversion plugin to convert markup from that ecosystem
-// to Pulumi, using the bridged providers.
+// `GetMappingRequest` is the type of requests sent as part of a [](pulumirpc.ResourceProvider.GetMapping) call.
 message GetMappingRequest {
-    // the conversion key for the mapping being requested.
+    // The conversion key for the mapping being requested. This typically corresponds to the source language, such as
+    // `terraform` in the case of mapping Terraform names to Pulumi names.
     string key = 1;
-    // the optional provider key for the mapping being requested, if this is empty the provider should assume this
-    // request is from an old engine from before GetMappings and should return it's "primary" mapping. If this is set
-    // then the `provider` field in GetMappingResponse should be the same.
+
+    // An optional *source provider key* for the mapping being requested. If this is empty, the provider should assume
+    // that this request is from an old engine prior to the introduction of [](pulumirpc.ResourceProvider.GetMappings).
+    // In these cases the request should be answered with the "primary" mapping. If this field is set, the `provider`
+    // field in the corresponding [](pulumirpc.GetMappingResponse) should contain the same value.
     string provider = 2;
 }
 
-// GetMappingResponse returns convert plugin specific data for this provider. This will normally be human
-// readable JSON, but the engine doesn't mandate any form.
+// `GetMappingResponse` is the type of responses sent by a [](pulumirpc.ResourceProvider.GetMapping) call. The data
+// within a `GetMappingResponse` will normally be human-readable JSON (e.g. an object mapping names from the source to
+// Pulumi), but the engine doesn't mandate any specific format.
 message GetMappingResponse {
-    // the provider key this is mapping for. For example the Pulumi provider "terraform-template" would return "template" for this.
+    // The *source provider key* that this mapping contains data for.
     string provider = 1;
 
-    // the conversion plugin specific data.
+    // Conversion-plugin-specific mapping data.
```

Comment:
Why dashes?

Reply:
It's a compound adjective, so I've hyphenated it (this might just be me though). E.g. the difference between "Working with big bearded people" (big people who have beards) and "Working with big-bearded people" (people who have big beards). That said, I'll avoid this and just reword it as "Mapping data specific to the conversion plugin" :sweat_smile: